### PR TITLE
Workaround for JsonRpc cancellation race condition.

### PR DESCRIPTION
**Pull request by tmat**
Use non-cancellable RPC invocations.

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi PTAL

**Comment by CyrusNajmabadi** [comment]
Looks totally fine to me.  Only question I would have is if this catches all the cases.  I'm basing this on your understanding of things and that these are the only entrypoints for invokable cancellation.

**Comment by tmat** [comment]
@CyrusNajmabadi Yes, these are the only ones.

The LSP also calls InvokeWithCancellationAsync but passes CancellationToken.None :-/
```
var result = await jsonRpc.InvokeWithCancellationAsync<JObject>(
                    Methods.InitializeName,
                    new object[] { new InitializeParams() },
                    CancellationToken.None);
```

**Comment by CyrusNajmabadi** [comment]
> The LSP also calls InvokeWithCancellationAsync but passes CancellationToken.None :-/

Note: it's unclear to me if that means we'll avoid the race condition.  i.e. it might go through some special JsonRpc path that is still susceptible (i honestly don't know).  Should we just update that guy as well to be on the safe side?

**Comment by tmat** [comment]
Not necessary, imo. The problem only happens when cancellation actually occurs during the call, not because the token is passed in.

**Comment by CyrusNajmabadi** [comment]
> Not necessary, imo. The problem only happens when cancellation actually occurs during the call, not because the token is passed in

Ok.  I defer to your judgment on that :)

**Comment by tmat** [comment]
I have replaced it. So that we have consistency.

File: src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs [hunk 1]
```diff
@@ -93,17 +93,17 @@ private async Task<ImmutableArray<SymbolInformation>> GetVsSearchResultsAsync(Te
 
             using (var jsonRpc = JsonRpc.Attach(await client.RequestServiceAsync(WellKnownServiceHubService.LanguageServer)))
             {
-                var result = await jsonRpc.InvokeWithCancellationAsync<JObject>(
+                // TODO: use InvokeWithCancellationAsync(targetName, arguments, cancellationToken)
+                // (https://github.com/dotnet/roslyn/issues/45980)
+                var result = await jsonRpc.InvokeAsync<JObject>(
                     Methods.InitializeName,
-                    new object[] { new InitializeParams() },
-                    CancellationToken.None);
```
Comment (sharwell) [inline src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs:99]: 📝 `InvokeAsync` just forwards to `InvokeWithCancellationAsync` using `CancellationToken.None`. Consider keeping the original code for all these calls.

Comment (tmat) [reply src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs:99]: OK

**Comment by jinujoseph** [comment]
https://github.com/microsoft/vs-streamjsonrpc/pull/492 to get this fixed.
Closing this as we won't need the workaround .

**Pull request closed without merging**